tests/lib/openstack_query/handlers/test_client_side_handler.py

```diff
@@ -0,0 +1,207 @@
+import unittest
+from unittest.mock import MagicMock, patch
+from parameterized import parameterized
+
+from openstack_query.handlers.client_side_handler import ClientSideHandler
+from exceptions.query_preset_mapping_error import QueryPresetMappingError
+
+from tests.lib.openstack_query.mocks.mocked_query_presets import MockQueryPresets
+from tests.lib.openstack_query.mocks.mocked_props import MockProperties
+
+
+class ClientSideHandlerBaseTests(unittest.TestCase):
+    """
+    Runs various tests to ensure that ClientSideHandlerBase class methods function expectedly
+    """
+
+    def setUp(self):
+        """
+        Setup for tests
+        """
+        super().setUp()
+
+        _FILTER_FUNCTION_MAPPINGS = {
+            MockQueryPresets.ITEM_1: ["*"],
+            MockQueryPresets.ITEM_2: [MockProperties.PROP_1, MockProperties.PROP_2],
+            MockQueryPresets.ITEM_3: [MockProperties.PROP_3, MockProperties.PROP_4],
+        }
+        self.instance = ClientSideHandler(_FILTER_FUNCTION_MAPPINGS)
+        self.instance._FILTER_FUNCTIONS = {
+            MockQueryPresets.ITEM_1: "item1_func",
+            MockQueryPresets.ITEM_2: "item2_func",
+            MockQueryPresets.ITEM_3: "item3_func",
+        }
+
+    def test_check_supported_true(self):
+        # when preset has '*' mapping - accepts all props
+        self.assertTrue(
+            self.instance.check_supported(
+                MockQueryPresets.ITEM_1, MockProperties.PROP_1
+            )
+        )
+
+        # when preset has explicit prop mapping
+        self.assertTrue(
+            self.instance.check_supported(
+                MockQueryPresets.ITEM_2, MockProperties.PROP_2
+            )
+        )
+
+    def test_check_preset_supported_false(self):
+        # when preset is not supported
+        self.assertFalse(
+            self.instance.check_supported(
+                MockQueryPresets.ITEM_4, MockProperties.PROP_1
+            )
+        )
+
+        # when preset found, but prop is not supported
+        self.assertFalse(
+            self.instance.check_supported(
+                MockQueryPresets.ITEM_2, MockProperties.PROP_3
+            )
+        )
+
+    @patch(
+        "openstack_query.handlers.client_side_handler.ClientSideHandler._check_filter_func"
+    )
+    @patch(
+        "openstack_query.handlers.client_side_handler.ClientSideHandler._filter_func_wrapper"
+    )
+    def test_get_filter_func_valid(
```

Comment:
Do the names need changing in the tests too?

Reply:
No they don't - we're testing `ClientSideHandler` object - its given that the filter function works only on the client side